Task 1

DrozdMary opened the pull request.

DrozdMary committed:
Init task_1

DrozdMary committed:
Merge remote-tracking branch 'origin/main' into task_1

# Conflicts:
#	.idea/Tasks.iml
#	.idea/misc.xml
#	.idea/vcs.xml

DrozdMary committed:
add in .gitignore folder .idea

DrozdMary committed:
Some little changes and refreshed requerements

DrozdMary committed:
added indexes for tables in db

DrozdMary committed:
delete idea

DrozdMary committed:
Merge remote-tracking branch 'origin/main' into task_1

# Conflicts:
#	.gitignore
#	.idea/workspace.xml

DrozdMary committed:
idea all files in folder

DrozdMary committed:
Merge branch 'main' into task_1

DrozdMary committed:
Delete .idea directory

DrozdMary committed:
try delete config.py

zaitsevv commented on task_1/src/db_connector.py:
```diff
@@ -0,0 +1,42 @@
+    def create_tables(self):
+        try:
+            self.cur.execute(
+                "create table  IF NOT EXISTS rooms ( id int NOT NULL, name varchar(50), primary key(id) )"
```
Соблюдай один код стайл - либо ты пишешь всё в нижнем регистре как `create table`, или все ключевые слова пишешь в верхнем как `IF NOT EXISTS`

zaitsevv commented on task_1/src/db_connector.py:
```diff
@@ -0,0 +1,42 @@
+import mysql.connector
+from config import host, user, password, db_name, port
+from logger import logger
```
Импорты из python core должны идти до импортов библиотек
```
from logger import logger

import mysql.connector
from config import ...
```

zaitsevv commented on task_1/src/db_loader.py:
```diff
@@ -0,0 +1,47 @@
+        placeholders = ", ".join(["%s"] * len(columns))
+        # последовательное добавления каждой записи в таблицу
+        try:
+            for items in data:
```
это работает, но можно проще, посмотри `json_populate_recordset`
![image](https://github.com/DrozdMary/Tasks/assets/82774766/dbccdf6e-c4f0-47df-9f5b-90ca12d7dd2c)

zaitsevv replied:
`{json}` это все данные из json файла в виде питон строки

zaitsevv replied:
или ещё можешь посмотреть в сторону `COPY`

DrozdMary replied:
да , намного проще, но это для PostgreSQL... в MySQL  "Not supported"(
можно я не буду менять на PostgreSQL?) а просто оставлю как есть ><

zaitsevv commented on task_1/src/db_connector.py:
```diff
@@ -0,0 +1,42 @@
+                "create table  IF NOT EXISTS rooms ( id int NOT NULL, name varchar(50), primary key(id) )"
+            )
+            self.cur.execute(
+                "create table  IF NOT EXISTS students( birthday datetime, id int primary key, name char(100), room int, "
```
лучше все строковые запросы вынести в отдельные файлы

zaitsevv commented on task_1/src/sql_script_executor.py:
```diff
@@ -0,0 +1,52 @@
+        converted_data = []
+        for item in data:
+            row = dict()
+            for i in range(len(columns)):
```
почитай про dict comprehension и перепиши этот цикл используя его

zaitsevv commented:
И добавь везде аннотации типов и docstrings на английском

DrozdMary committed:
necessary changes
(added docstrings and anotations, put all the scripts in a file, change to dict comprehension, changed the order of imports, )
